CDTDatastore/CDTReplicator/CDTPullReplication.m

```diff
@@ -33,7 +34,16 @@ + (instancetype)replicationWithSource:(NSURL *)source target:(CDTDatastore *)tar
 - (instancetype)initWithSource:(NSURL *)source target:(CDTDatastore *)target
 {
     if (self = [super init]) {
-        _source = source;
+        
+        NSURLComponents * sourceComponents = [NSURLComponents componentsWithURL:source resolvingAgainstBaseURL:NO];
```

Comment:
I agree that this duplication is unfortunate, a couple of thoughts:

If we add a parameter to the superclass `init` could we just create/add the interceptor there and avoid the confusion (of reading the subclass in isolation) by storing the sanitized `NSURL` in the superclass and then using it as the target or source as appropriate in the subclasses?

or

Could we avoid the awkward dictionary/array return type by just having two separate class methods, one which returns a `CDTSessionCookieInterceptor` and one which returns a sanitized `NSURL`?